src/pages/students/[student_id].tsx

```diff
@@ -1,44 +1,85 @@
 import { trpc } from "@/client/lib/trpc";
+import EditStudentRow from "@/components/editStudentRow/EditStudentRow";
 import $button from "@/styles/Button.module.css";
 import $home from "@/styles/Home.module.css";
 import $input from "@/styles/Input.module.css";
 import Box from "@mui/material/Box";
 import Button from "@mui/material/Button";
 import Container from "@mui/material/Container";
 import Modal from "@mui/material/Modal";
+import Stack from "@mui/material/Stack";
+import { addYears, format, parseISO, subDays } from "date-fns";
 import Image from "next/image";
-import Link from "next/link";
 import { useRouter } from "next/router";
 import { useState } from "react";
 import Iep from "../../components/iep/Iep";
 import noGoals from "../../public/img/no-goals-icon.png";
 import $Form from "../../styles/Form.module.css";
+import $Image from "../../styles/Image.module.css";
 import $Modal from "../../styles/Modal.module.css";
 import $StudentPage from "../../styles/StudentPage.module.css";
-import $Image from "../../styles/Image.module.css";
-import { parseISO, addYears, subDays, format } from "date-fns";
 
 const ViewStudentPage = () => {
   const [createIepModal, setCreateIepModal] = useState(false);
   const [archivePrompt, setArchivePrompt] = useState(false);
   const [startDate, setStartDate] = useState("");
   const [endDate, setEndDate] = useState("");
+  const [viewState, setViewState] = useState(0);
 
   const utils = trpc.useContext();
   const router = useRouter();
   const { student_id } = router.query;
 
+  const VIEW_STATES = { MAIN: 0, EDIT: 1 };
+
+  const handleEditState = () => {
+    setViewState(VIEW_STATES.EDIT);
+  };
+
+  const handleMainState = () => {
+    setViewState(VIEW_STATES.MAIN);
+  };
+
   const { data: student, isLoading } = trpc.student.getStudentById.useQuery(
     { student_id: student_id as string },
     { enabled: Boolean(student_id) }
   );
 
+  const buttonSX = {
+    "&:hover": {
+      background: "#3023B8",
+    },
+  };
+
+  const [firstName, setFirstName] = useState(student?.first_name || "");
+  const [lastName, setLastName] = useState(student?.last_name || "");
+  const [email, setEmail] = useState(student?.email || "");
+  const [grade, setGrade] = useState(student?.grade || 0);
```

Comment:
instead of making all inputs controlled, I would recommend instead wrapping with a `<form/>` and using uncontrolled inputs

`<form/>` is also preferred as it's slightly better for accessibility

see `src/components/goal/Goal.tsx` for an example

Reply:
Great advice! I refactored implemented the uncontrolled inputs using form suggestion, and removed lines 54-57